File: backend/src/main/java/kg/taskflow/service/NotificationSettingsService.java
package kg.taskflow.service;

import kg.taskflow.dto.notification.NotificationSettingsDto;
import kg.taskflow.dto.notification.TelegramLinkDto;
import kg.taskflow.dto.notification.TelegramStatusDto;
import kg.taskflow.dto.notification.UpdateNotificationSettingsRequest;

public interface NotificationSettingsService {
    NotificationSettingsDto getMySettings();
    NotificationSettingsDto updateSettings(UpdateNotificationSettingsRequest request);
    TelegramStatusDto getTelegramStatus();
    TelegramLinkDto generateTelegramLink();
    void unlinkTelegram();
}

File: frontend/src/services/notification-settings.service.ts
import api from './api';
import {
  NotificationSettings,
  UpdateNotificationSettingsRequest,
  TelegramStatus,
  TelegramLink,
} from '@/types/notification-settings.types';

export const notificationSettingsService = {
  async getSettings(): Promise<NotificationSettings> {
    const response = await api.get<NotificationSettings>('/notification-settings');
    return response.data;
  },

  async updateSettings(request: UpdateNotificationSettingsRequest): Promise<NotificationSettings> {
    const response = await api.put<NotificationSettings>('/notification-settings', request);
    return response.data;
  },

  async getTelegramStatus(): Promise<TelegramStatus> {
    const response = await api.get<TelegramStatus>('/notification-settings/telegram/status');
    return response.data;
  },

  async generateTelegramLink(): Promise<TelegramLink> {
    const response = await api.get<TelegramLink>('/notification-settings/telegram/link');
    return response.data;
  },

  async unlinkTelegram(): Promise<void> {
    await api.delete('/notification-settings/telegram');
  },
};
